packages/test-runner-saucelabs/src/createSauceLabsLauncher.ts

```diff
@@ -33,39 +35,56 @@ export function createSauceLabsLauncher(
   }
   const manager = new SauceLabsLauncherManager(finalSauceLabsOptions, finalConnectOptions);
 
-  return function sauceLabsLauncher(capabilities: Record<string, unknown>): BrowserLauncher {
+  return function sauceLabsLauncher(capabilities: DesiredCapabilities): BrowserLauncher {
     if (capabilities == null) {
       throw new Error('Capabilities are required.');
     }
-    const finalCapabilities = { ...capabilities };
 
-    // sync the tunnel identifier, username and access key
-    finalCapabilities['sauce:options'] = {
-      ...(finalCapabilities['sauce:options'] as Record<string, unknown>),
-      username: saucelabsOptions.user,
-      accessKey: saucelabsOptions.key,
+    let finalCapabilities = { ...capabilities };
+
+    const finalSauceCapabilities = {
       tunnelIdentifier: finalConnectOptions.tunnelIdentifier,
+      ...saucelabsCapabilities,
+    };
+
+    // W3C capabilities: only browserVersion is mandatory, platformName is optional.
+    // Note that setting 'sauce:options' forces Sauce Labs to use W3C capabilities.
+    if (capabilities.browserVersion) {
+      finalCapabilities.platformName =
+        finalCapabilities.platformName || finalCapabilities.platform || 'Windows 10';
+      finalCapabilities['sauce:options'] = {
+        ...finalSauceCapabilities,
+        ...(finalCapabilities['sauce:options'] || {}),
+      };
+
+      // Delete JWP capabilities, if any.
+      delete finalCapabilities.version;
+      delete finalCapabilities.platform;
```

Comment:
Why do we need to delete these?

Reply:
Updated to document that non-W3C capabilities need to be removed.